openlibrary/macros/OLID.html

```diff
@@ -0,0 +1,15 @@
+$def with (q="")
+
+$ edition = query_param('edition', q)
+$ result = edition.olid_link(q)
```

Comment:
@psyren99:

`edition` is not defined here. You defined `olid_link` on a class called `Edition`, and `Edition` is not accessible in this template.

You need to either (a) add a function into the template globals, e.g. https://github.com/internetarchive/openlibrary/blob/master/openlibrary/plugins/openlibrary/code.py#L1358-L1385 or (b) define a `@public` function that is accessible to the template.

See https://github.com/internetarchive/openlibrary/blob/master/openlibrary/plugins/upstream/code.py#L294-L305... In fact which basically does what we want already (*and* follows redirects)